### ISSUE TITLE
Very short TE EPIs are challenging for BBR

### ISSUE THREAD

**Issue by oesteban** [poldracklab/fmriprep#997]
DS000148 has a TE of 18ms, what yields a contrast like:
<img width="383" alt="screen shot 2018-02-20 at 12 28 00 pm" src="https://user-images.githubusercontent.com/598470/36448308-0e73e0a2-163c-11e8-8bb7-8e3c1caa4c5b.png">

This makes it very challenging for BBR to find the WM/GM boundary.

**Comment by effigies** [comment]
Looks like a pretty intense bias field, too. How does the enhance and skull strip do?

**Comment by chrisfilo** [comment]
Would be useful to see the contrast enhanced and bias corrected reference volume that BBR "sees".

**Comment by oesteban** [comment]
This is one of the few runs that worked in sub-09 (run-02):
<img width="545" alt="ds148-bbr" src="https://user-images.githubusercontent.com/598470/36450524-0e08a5e2-1643-11e8-9152-2f53847916e8.png">

Most of the runs fell back to the `mri_coreg` initialization. As you can see, the 9 parameters caused that bbregister expanded the image.

Two things we should consider:
- if SDC was run, then limit the BBR to 6 parameters (don't allow those 3 free parameters to overfit an inexistent deformation after SDC).
- for SDC-SyN: when the TE is this short and the intensity of CSF is below the WM and GM then bypass the T1 inversion.

Other than that, taking into consideration the original contrast, I don't think the reference generation is performing badly.

**Comment by oesteban** [comment]
Please note that in the above picture I made sure the reportlet was 100% at the "moving" frame.

**Comment by oesteban** [comment]
@effigies I see what you meant before about the inhomogeneity, would you agree that it is not present anymore in the reference?

**Comment by effigies** [comment]
Yes, but is that from the same series as the first image in this thread?

**Comment by oesteban** [comment]
This would be DS148/sub-09 (run-02), the [same as in my previous comment](https://github.com/poldracklab/fmriprep/issues/997#issuecomment-367129621).
<img width="385" alt="screen shot 2018-02-20 at 3 36 11 pm" src="https://user-images.githubusercontent.com/598470/36455172-d494a958-1653-11e8-802c-a0ec1b1e6bb9.png">

**Comment by oesteban** [comment]
Less INU, but still present

**Comment by oesteban** [comment]
@effigies, what do you think about not inverting the T1 in SDC-SyN for these kind of images?

**Comment by effigies** [comment]
Okay. So that looks like we see an improvement in contrast, though slice effects are also strongly enhanced.

So the argument for not inverting the T1 is just that the white/CSF contrast will dominate? I guess I'm not sure that we can expect SyN to work at all well on an image with such low contrast in the first place. Might be worth trying, but I suspect that it's not going to be good either way you do it.

One thing that might be worth returning to is the idea of using a T2w image in place of the inverted T1w image, when available. That should keep the CSF contrast ratio sensible while still allowing what little gray/white contrast there is in these images to constrain the SyN.

**Comment by oesteban** [comment]
If I understand correctly why the T1w contrast is inverted at all, whenever we have a T2w that would be preferable over the T1w-inverted, correct?

Moreover: the rationale behind inverting the T1w is that the contrast distribution is closer to that of a BOLD (T2*) contrast right? However these short TE images have more T1 contrast and less T2 contrast. That's why I was suggesting this. For these images, it seems to me that using the T2w (when available) would be as bad as inverting the intensity of the T1w, is it reasonable?.

> I guess I'm not sure that we can expect SyN to work at all well on an image with such low contrast in the first place. 

I will need to look at what metric we are using. It would not make a lot of sense to invert the T1 to apply some information-theory based metric. I guess the latest registration stage is done with a cross-correlation or similar. The ratios between the intensities of tissues need to be taken cautiously because we are most likely enabling the histogram matching of the inputs (and that changes those ratios).

Coming back to the original issue, I think we all can agree on the fact that this contrast is not good at all for BBR, is that correct?

**Comment by effigies** [comment]
> If I understand correctly why the T1w contrast is inverted at all, whenever we have a T2w that would be preferable over the T1w-inverted, correct?

Yes.

> Moreover: the rationale behind inverting the T1w is that the contrast distribution is closer to that of a BOLD (T2*) contrast right? However these short TE images have more T1 contrast and less T2 contrast. That's why I was suggesting this. For these images, it seems to me that using the T2w (when available) would be as bad as inverting the intensity of the T1w, is it reasonable?.

Hmm, maybe. It's a reasonable argument, but from these images it doesn't look like we're getting anything like a T1w gray/white contrast, here, so it's hard for me to say. I think I'd have to look at histograms for T1w, T2w and short TE BOLD series. And it might help to be a little more awake and less hungry. :-)

> Coming back to the original issue, I think we all can agree on the fact that this contrast is not good at all for BBR, is that correct?

It definitely doesn't look good to me. Especially with 9-DOF fitting, we're asking for trouble.

**Comment by oesteban** [comment]
Let's address BBR and short-TE exclusively here, and the tangents about SDC in #998.

**Comment by chrisfilo** [comment]
One more from the same dataset:
![image](https://user-images.githubusercontent.com/238759/36930212-46451a1a-1e51-11e8-8cfc-a43193793756.png)
s000148/reportlets/fmriprep/sub-26/func/sub-26_task-figure2backwith1backlures_run-01_bold_bbr.svg

seems like short TE cause inverted contrast in this bold image (ventricles are dark, but should be bright)
